cvmfs/ssl.h

```diff
@@ -0,0 +1,61 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#ifndef CVMFS_SSL_H_
+#define CVMFS_SSL_H_
+
+#include <string>
+#include <vector>
+
+#include "duplex_curl.h"
+#include "util/posix.h"
+
+#ifdef CVMFS_NAMESPACE_GUARD
+namespace CVMFS_NAMESPACE_GUARD {
+#endif
+
+inline unsigned int count_ssl_certificates(std::string directory) {
+  if (!DirectoryExists(directory)) return 0;
+
+  std::vector<std::string> dotpem = FindFilesBySuffix(directory, ".pem");
+  std::vector<std::string> dotcrt = FindFilesBySuffix(directory, ".crt");
+
+  return dotpem.size() + dotcrt.size();
+}
+
+inline bool AddSSLCertificates(CURL *handle) {
+  std::vector<std::string> cadirs;
+  const char *cadir = getenv("X509_CERT_DIR");
+  if (cadir != NULL) {
+    cadirs.push_back(std::string(cadir));
+  }
+  cadirs.push_back("/etc/grid-security/certificates");
+
+  // most systems store the certificates here
+  cadirs.push_back("/etc/ssl/certs/");
+
+  cadirs.push_back("/etc/pki/tls/certs/");
+  cadirs.push_back("/etc/ssl/");
+  cadirs.push_back("/etc/pki/tls/");
+  cadirs.push_back("/etc/pki/ca-trust/extracted/pem/");
+  cadirs.push_back("/etc/ssl/");
+
+  for (std::vector<std::string>::const_iterator cadir = cadirs.begin();
+       cadir != cadirs.end(); ++cadir) {
+    if (count_ssl_certificates(*cadir) > 0) {
```

Comment:
Perhaps it makes more sense to call it `HasCertificates`.  Then we can back out if we find at least one, we don't need to count them all.

Reply:
I did change the name but the implementation is the same. Even if we want to check just one, is still is the easiest thing to do.

Should we change also the implementation? It will anyway be a couple of stat calls against the directory, we might save some string comparison with an early return.

Reply:
Probably OK for now because libcurl anyway has to read them all.  Still, there can be potentially a lot of certificate files, so we should at least add a `TODO` comment to the implementation to keep note of a possible improvement.

Reply:
Hummm, I prefer to do it now. But I am not sure I see any potential improvement.

Reply:
I'm thinking of this: suppose we look into a directory with 10,000 files.  If the first of them is a .pem or a .crt file, we don't need to stat and check file names on the 9999 other files.

Reply:
Alright, let's wait anyway to merge this.